spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,387 @@
+describe EquipmentModelsController, type: :controller do
+  before(:each) { mock_app_config }
+  def mock_eq_model(**attrs)
+    instance_spy('EquipmentModel', **attrs)
   end
 
-  it 'should limit @associated_equipment_models to maximum 6' do
-    mod1 = FactoryGirl.create(:equipment_model)
-    mod2 = FactoryGirl.create(:equipment_model)
-    mod3 = FactoryGirl.create(:equipment_model)
-    mod4 = FactoryGirl.create(:equipment_model)
-    mod5 = FactoryGirl.create(:equipment_model)
-    mod6 = FactoryGirl.create(:equipment_model)
-    mod7 = FactoryGirl.create(:equipment_model)
-    model.associated_equipment_models =
-      [mod1, mod2, mod3, mod4, mod5, mod6, mod7]
-    get :show, id: model
-    expect(assigns(:associated_equipment_models).size).to eq(6)
+  def mock_eq_model_findable(**attrs)
+    id = FactoryGirl.generate(:unique_id)
+    model = mock_eq_model(id: id)
+    allow(EquipmentModel).to receive(:find).with(id).and_return(model)
+    model
   end
-end
 
-shared_examples_for 'GET index success' do
-  it { is_expected.to respond_with(:success) }
-  it { is_expected.to render_template(:index) }
-  it { is_expected.not_to set_flash }
-  context 'without show deleted' do
-    let!(:mod_other_cat_active) { FactoryGirl.create(:equipment_model) }
-    let!(:mod_other_cat_inactive) do
-      FactoryGirl.create(:equipment_model,
-                         deleted_at: Time.zone.today)
-    end
-    context 'with @category set' do
-      it 'should populate an array of of active category-type equipment '\
-        'models' do
-        mod_same_cat_inactive =
-          FactoryGirl.create(:equipment_model, category: model.category,
-                                               deleted_at: Time.zone.today)
-        get :index, category_id: model.category
-        expect(assigns(:equipment_models).include?(model)).to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_active))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).include?(mod_same_cat_inactive))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_inactive))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).size).to eq(1)
-      end
-    end
-    context 'without @category set' do
-      it 'should populate an array of all active equipment models' do
-        expect(assigns(:equipment_models).include?(model)).to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_active))
-          .to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_inactive))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).size).to eq(2)
-      end
-    end
+  def mock_eq_collection(models: nil, count: 1)
+    models ||= Array.new(count) { mock_eq_model }
+    allow(EquipmentModel).to receive(:all).and_return(models)
+    allow(EquipmentModel).to receive(:where).and_return([])
```

Comment:
this `where` stub is really more of a workaround and should maybe be reconsidered